merge_sorted_array.py

```diff
@@ -0,0 +1,69 @@
+# 两个list nums1 和 nums2 都是sorted list
+# 但是 nums1 后半部分可能不是, 只不过留出空间给nums2
+# 要求是把nums2合并到nums1并保持排序状态
+# 只能通过更改nums1, 不能创造任何新的list, 不需要return任何值
+
+# 具体问题请参考test case的例子.
+
+# 另外:
+# 改变一个list的方法有lst.pop, lst.insert, lst.remove, 或者直接lst[i] = x.
+
+def merge(nums1, m, nums2, n):
+    """
+    :type nums1: List[int]
+    :type m: int
+    :type nums2: List[int]
+    :type n: int
+    :rtype: void Do not return anything, modify nums1 in-place instead.
+    """
+    # your code:
+
```

Comment:
cur = 0
    tmp_n= 0
    for j in nums2:
        while j>=nums1[cur] and cur < m+tmp_n:
            cur += 1
        nums1.insert(cur, j)
        tmp_n += 1
        nums1.pop()
    for i in range(m+n,len(nums1)):
        nums1[i] = 0